### PR TITLE
Modules stub

### PR DESCRIPTION
**Pull request by LouisLeNezet**
## PR checklist

Add stub section to multiple modules

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/glimpse/chunk/main.nf [hunk 1]
```diff
@@ -31,7 +31,19 @@ process GLIMPSE_CHUNK {
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
-        glimpse: "\$(GLIMPSE_phase --help | sed -nr '/Version/p' | grep -o -E '([0-9]+.){1,2}[0-9]')"
+        glimpse: "\$(GLIMPSE_chunk --help | sed -nr '/Version/p' | grep -o -E '([0-9]+.){1,2}[0-9]')"
```
Comment (JoseEspinosa) [inline modules/nf-core/glimpse/chunk/main.nf:34]: Good spot :heart_eyes:

File: modules/nf-core/glimpse2/phase/main.nf [hunk 2]
```diff
@@ -69,4 +69,23 @@ process GLIMPSE2_PHASE {
...
+
+    stub:
+    def region = input_region    ? "${output_region.replace(":","_")}" : "${reference}"
+    def args   = task.ext.args   ?: ""
+    def prefix = task.ext.prefix ?: "${meta.id}_${region}"
+    def suffix = task.ext.suffix ?: "bcf"
+    """
+    if $input_bam
+    then
+        touch all_bam.txt
```
Comment (JoseEspinosa) [inline modules/nf-core/glimpse2/phase/main.nf:81]: Is this needed? I don't see it declared as an `output`

File: modules/nf-core/glimpse2/splitreference/main.nf [hunk 3]
```diff
@@ -30,9 +30,9 @@ process GLIMPSE2_SPLITREFERENCE {
         task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ''
-    def prefix = task.ext.prefix ?: "${meta.id}_${output_region.replace(":","_")}"
-    def map_command = map ? "--map $map" : ""
+    def args        = task.ext.args   ?: ''
```
Comment (JoseEspinosa) [inline modules/nf-core/glimpse2/splitreference/main.nf:33]: :heart:

File: modules/nf-core/shapeit5/ligate/main.nf [hunk 4]
```diff
@@ -31,8 +31,22 @@ process SHAPEIT5_LIGATE {
...
+        shapeit5: "\$(SHAPEIT5_ligate | sed -nr '/Version/p' | grep -o -E '([0-9]+.){1,2}[0-9]' | head -n 1)"
+    END_VERSIONS
+    """
+
+    stub:
+    def args   = task.ext.args   ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def suffix = task.ext.suffix ?: "vcf.gz"
+    """
+    touch all_files.txt
```
Comment (JoseEspinosa) [inline modules/nf-core/shapeit5/ligate/main.nf:44]: Again I dont see these file declared as output, so why you need to create it?

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
JVery nice!  :rocket: 
Just two small comments probably because I missed something

File: modules/nf-core/glimpse2/phase/main.nf (hunk 2 above)
Comment (JoseEspinosa) [reply modules/nf-core/glimpse2/phase/main.nf:81]: You only need to generate the files that are declared as `output` as those are the only staged by the process and hence will be the only validated by any test or expected during the `stub` run. I mean, it doesn't harm that they are generated but just not necessary.

Comment (LouisLeNezet) [reply modules/nf-core/glimpse2/phase/main.nf:81]: Okay !

**Review by JoseEspinosa — APPROVED** [review]
Awesome! 😍

**Pull request merged**